automol/form/_form.py

```diff
@@ -131,57 +121,47 @@ def add_element(fml, symb, num=1):
     return fml
 
 
-def join(fml1, fml2):
+def join(fml1: dict[str:int], fml2: dict[str:int]) -> int:
```

Comment:
Minor syntax correction: `dict[str, int]`.

Also, the return value should be another formula, `-> dict[str, int]`.